templates/powershell/Get-AdoCallBackApiToken.ps1

```diff
@@ -0,0 +1,82 @@
+<#
+.SYNOPSIS
+Get keyvault secret.
+
+.DESCRIPTION
+Get keyvault secret and store it in task variable.
+
+.PARAMETER KeyVaultName
+Mandatory. KeyVault Name.
+
+.PARAMETER -TenantId
+Mandatory. Tenant Id.
+
+
+.EXAMPLE
+.\Get-KeyVaultSecret KeyVaultName <KeyVaultName> -TenantId <TenantId> 
+#> 
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory)] 
+    [string]$KeyVaultName,
+    [Parameter(Mandatory)] 
+    [string]$TenantId
+)
+
+Set-StrictMode -Version 3.0
+
+[string]$functionName = $MyInvocation.MyCommand
+[datetime]$startTime = [datetime]::UtcNow
+
+[int]$exitCode = -1
+[bool]$setHostExitCode = (Test-Path -Path ENV:TF_BUILD) -and ($ENV:TF_BUILD -eq "true")
+[bool]$enableDebug = $false
+
+Set-Variable -Name ErrorActionPreference -Value Continue -scope global
+Set-Variable -Name InformationPreference -Value Continue -Scope global
+
+if ($enableDebug) {
+    Set-Variable -Name VerbosePreference -Value Continue -Scope global
+    Set-Variable -Name DebugPreference -Value Continue -Scope global
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:KeyVaultName=$KeyVaultName"
+Write-Debug "${functionName}:TenantId=$TenantId"
+
+try {
+    Write-Host "Fetching Keyvault secrets from KeyVaultName $($KeyVaultName)"
+    [string]$AdoCallBackApiClientAppClientId = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name "ADOCALLBACK-API-CLIENT-APP-REG-CLIENT-ID" -AsPlainText -ErrorAction Stop
```

Comment:
actually don't use these commands for get-secrets.. we've deprecated usage. Use the Get  KV Secrets task please. Ping Ajay for context but this is a risky command that exposes secrets

Reply:
@DaRoszko - Also just got hold of @ajaysoni-cap and he is happy with this approach to solve the problem of outputting secrets when the pipeline is ran in debug mode.